components/ui/input/index.tsx

```diff
@@ -0,0 +1,287 @@
+'use client';
+import React, { useMemo } from 'react';
+import { createInput } from '@gluestack-ui/input';
+import { Svg } from 'react-native-svg';
+import { View, Pressable, TextInput, Platform } from 'react-native';
+import { tva } from '@gluestack-ui/nativewind-utils/tva';
+import {
+  withStyleContext,
+  useStyleContext,
+} from '@gluestack-ui/nativewind-utils/withStyleContext';
+import { withStyleContextAndStates } from '@gluestack-ui/nativewind-utils/withStyleContextAndStates';
+import { cssInterop } from 'nativewind';
+import { withStates } from '@gluestack-ui/nativewind-utils/withStates';
+import type { VariantProps } from '@gluestack-ui/nativewind-utils';
+const SCOPE = 'INPUT';
+
+type IPrimitiveIcon = {
+  height?: number | string;
+  width?: number | string;
+  fill?: string;
+  color?: string;
+  size?: number | string;
+  stroke?: string;
+  as?: React.ElementType;
+  className?: string;
+  classNameColor?: string;
+};
+
+const PrimitiveIcon = React.forwardRef<
+  React.ElementRef<typeof Svg>,
+  IPrimitiveIcon
+>(
+  (
+    {
+      height,
+      width,
+      fill,
+      color,
+      classNameColor,
+      size,
+      stroke = 'currentColor',
+      as: AsComp,
+      ...props
+    },
+    ref
+  ) => {
+    color = color ?? classNameColor;
+    const sizeProps = useMemo(() => {
```

Comment:
instead of having these several `ifs` you can do something like this. 
```
useMemo(
    () => ({
        ...(size && {size}),
        ...(height && {height}),
        ...(width && {width}),
    }),
    [size, height, width]
)
```

Reply:
Forgot to mention that this file and all the files inside /components/ui folder are from gluestack-ui, not mine. They are added when I want to use any of these library components.